source/tests/llassetgen-tests/Packing.cpp

```diff
@@ -0,0 +1,112 @@
+#include <gmock/gmock.h>
+
+#include <llassetgen/llassetgen.h>
+
+using llassetgen::Packing;
+
+using Vec = llassetgen::Vec2<llassetgen::PackingSizeType>;
+
+/**
+ * Base class for all packing test fixtures.
+ *
+ * Defines several test methods that should be run for all packing methods.
+ */
+class PackingTest : public testing::Test {
+   protected:
+    virtual bool run(const std::vector<Vec>& rectSizes, bool allowRotations, Packing& packing) = 0;
+    virtual Packing run(const std::vector<Vec>& rectSizes, bool allowRotations) = 0;
+
+    static bool rotatedSizesEquals(Vec size1, Vec size2) { return size1 == size2 || size1 == Vec{size2.y, size2.x}; }
+
+    /**
+     * Expect a fixed size packing to succeed and validate the result.
+     */
+    void expectSuccessfulValidPacking(const std::vector<Vec>& rectSizes, Vec atlasSize, bool allowRotations);
+
+    /**
+     * Expects a variable size packing to return a valid result.
+     */
+    void expectValidPacking(const std::vector<Vec>& rectSizes, bool allowRotations) {
+        validatePacking(run(rectSizes, allowRotations), rectSizes, allowRotations);
```

Comment:
Why is this function defined here and not in lines 72ff, as the other functions?

Reply:
I tend to write very short functions inside the class body just to keep the code shorter, but I don't have really have a strong opinion on what the best style for that is.

Reply:
I understand your reasoning.
The best style though is Consistency. If it were just one function inside expectValidPacking (like "run"), then it would be okay for me since it's only "forwarding". This function might be short written, but there is a lot that happens here.
When would it be "long enough" to define that functions at 72ff? There is no rule for that, rather opinions.
I'd really like you to be consistent.